presentation/src/main/java/com/safetyheads/akademiaandroida/presentation/ui/fragments/changepassword/ChangePasswordFragment.kt

```diff
@@ -0,0 +1,72 @@
+package com.safetyheads.akademiaandroida.presentation.ui.fragments.changepassword
+
+import android.os.Bundle
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewGroup
+import androidx.fragment.app.Fragment
+import com.safetyheads.akademiaandroida.presentation.R
+import com.safetyheads.akademiaandroida.presentation.databinding.FragmentChangePasswordBinding
+import com.safetyheads.akademiaandroida.presentation.ui.components.snackbar.LoginSnackBar
+import com.safetyheads.akademiaandroida.presentation.ui.utils.PasswordValidator
+import org.koin.androidx.viewmodel.ext.android.viewModel
+
+class ChangePasswordFragment : Fragment() {
+    private lateinit var binding: FragmentChangePasswordBinding
+    private val changePasswordViewModel: ChangePasswordViewModel by viewModel()
+
+    override fun onCreateView(
+        inflater: LayoutInflater, container: ViewGroup?,
+        savedInstanceState: Bundle?
+    ): View {
+        binding = FragmentChangePasswordBinding.inflate(inflater, container, false)
+        return binding.root
+    }
+
+    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
+
+        val etCurrentPassword = binding.editTextCurrentPassword
+        val etNewPassword = binding.editTextNewPassword
+        val etConfirmNewPassword = binding.editTextConfirmNewPassword
+        val savePassword = binding.savePassword
+        val rejectPassword = binding.rejectPassword
+
+        PasswordValidator.attach(etCurrentPassword, requireContext())
+        PasswordValidator.attach(etNewPassword, requireContext())
+        PasswordValidator.attach(etConfirmNewPassword, requireContext())
+
+        changePasswordViewModel.isSuccess.observe(viewLifecycleOwner) { isSuccess ->
+            if (isSuccess) {
+                LoginSnackBar.make(
+                    binding.root,
+                    message = requireActivity().getString(R.string.reset_password_success)
+                ).show()
+            } else {
+                LoginSnackBar.make(
+                    binding.root,
+                    message = requireActivity().getString(R.string.failed_reset_password)
+                ).show()
+            }
+        }
+
+        rejectPassword.setOnClickListener {
+            etCurrentPassword.setText("")
+            etNewPassword.setText("")
+            etConfirmNewPassword.setText("")
+        }
+
+        savePassword.setOnClickListener {
+            if (etCurrentPassword.error != "" && etNewPassword.error != "" && etConfirmNewPassword.error != "") {
```

Comment:
I think that you can check nulls here. You can use extension method isCorrectText() in EditText.kt file in project

Reply:
Error are never nulls. EditText doesnt have an error it has empty string

Reply:
I mean EditText.text when empty. If we click the button, the application will ask the database to change the password with a blank String passwords. I think it's good to specify it and add a check if it's not empty because we unnecessarily ask the database to return an failure to show a Snackbar error information

Reply:
Ok i disable save button for some cases